packages/db/src/types/granule.ts

```diff
@@ -1,7 +1,10 @@
-export interface PostgresGranule {
+export interface PostgresGranuleUniqueColumns {
```

Comment:
If we _do_ go with this approach of having a type specifically for identifying required vs. optional properties, maybe we could make it more generic.

I was looking at this:

https://stackoverflow.com/questions/52984808/is-there-a-way-to-get-all-required-properties-of-a-typescript-object

If we had a generic type `RequiredKeys` maybe we could use that where we want to validate against it rather than defining an entirely new type. I haven't 100% thought this out but the base `delete` signature would be something like:

```
async delete(
    knexOrTransaction: Knex | Knex.Transaction,
    params: RequiredKeys<RecordType>
  )
```

Reply:
I think the problem with that approach is that the subset of required keys for writing a granule is a larger subset than the subset of keys for uniquely identifying a granule.

So the "required" keys for a granule (before write) are `granule_id`, `collection_cumulus_id`, **and `status`**. However, we don't want to force anyone to specify `status` to uniquely identify a granule/delete it. 

The other problem is that the set of required keys for uniquely identifying a data type (granule, collection, etc) is going to be different per type and I don't think we can even write a generic helper to figure out it automatically. Even if we could, the implementation of a generic `RequiredKeys` solution in that post is pretty messy. And I think we would lose clarity/maintainability vs just being explicit.